src/client/App/App.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { fetchItems } from './actions';
+import Footer from '../components/Footer';
+import Header from '../components/Header';
+import MessageBox from '../components/MessageBox';
+import ProductsContainer from '../components/ProductsContainer';
+import Preloader from '../components/Preloader';
+import InputBar from '../components/InputBar/index';
+
+class App extends Component {
+  componentDidMount() {
+    this.fetchItems();
+  }
+
+  fetchItems = () => {
+    const { fetchItems } = this.props;
+
+    fetchItems();
+  };
+
+  render() {
+    const { fetchStatus, items } = this.props;
+    const archiveList = items.filter(item => item.isOrdered).reverse();
+    const shoppingList = items.filter(item => !item.isOrdered).reverse();
+
+    return (
+      <div
+        className={classNames('app-wrapper', {
+          overlay: fetchStatus === 'true'
+        })}
+      >
+        <Header />
+        {fetchStatus === 'error' && (
+          <MessageBox
+            message="Fetching failed. Try to refresh the page."
+            type="error"
+          />
+        )}
+
+        <InputBar />
+        <ProductsContainer products={shoppingList} />
+        <ProductsContainer isArchive products={archiveList} />
+        <Footer />
+        {fetchStatus === 'true' && <Preloader message="Fetching data..." />}
+      </div>
+    );
+  }
+}
+
+App.propTypes = {
+  fetchStatus: PropTypes.string,
```

Comment:
I've been talking with Kuba about this topic because we need 2 statuses: one for fetching data, second for adding new items. 
We decided that for representation three states of status which is pending, resolved and error we gonna use strings instead of booleans.

![screenshot from 2018-12-18 10-46-05](https://user-images.githubusercontent.com/27632432/50145701-71e86280-02b2-11e9-8a47-9ef0e6f1dd7a.png)

But maybe there is a better way to store these statuses in the state. 

```
fetchStatus: boolean(true,false),
newItemStatus: boolean(true, false),

err: Object {
    status: boolean(true,false),
    message: String(eg. "Fetching went wrong", "Adding new item went wrong");
}
```

Something like this?? Two statuses properties and one err property as an object with some error status and error message??  Drop me a line what would be best solution.

Reply:
Please just create 'enum' with consts so that we don't have string comparison. You can easily make a typo in string comparison.

Reply:
Please create prop type for that enum